synapse/lib/cell.py

```diff
@@ -196,6 +196,14 @@ def setCellUser(self, iden):
         await self._reqUserAllowed(*perm)
         return await self.cell.hive.pop(path)
 
+    @adminapi
+    async def saveHiveTree(self, path=()):
+        return await self.cell.hive.saveHiveTree(path=path)
```

Comment:
A user with `hive:get` permission can collect this same data if they can access the requested path. Why does this need to be marked as `@adminapi` and not checking the following:

```
perm = ('hive:get',) + path
await self._reqUserAllowed(*perm)
```

?